File: problems/Reverse Vowels of a String/reverseVowels.cs
public class Solution {
    public string ReverseVowels(string s) {
        var store = new HashSet<char>(new []{'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'});
        var stack = new Stack<char>();
        var queue = new Queue<char>();
        var sb = new StringBuilder();

        foreach (var item in s) {
            if (store.Contains(item)) {
                queue.Enqueue('$');
                stack.Push(item);
            } else {
                queue.Enqueue(item);
            }
        }

        foreach (var item in queue) {
            if ('$' == item) {
                sb.Append(stack.Pop());
            } else {
                sb.Append(item);
            }
        }

        return sb.ToString();
    }
}

File: problems/Reverse Vowels of a String/reverseVowels.java
class Solution {
    public String reverseVowels(String s) {
        Set<Character> vowels = Set.of('a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U');
        char[] arr = s.toCharArray();
		int i = 0;
		int j = arr.length - 1;

		while (i <= j) {
			if (vowels.contains(arr[i]) && vowels.contains(arr[j])) {
				swap(arr, i, j);
				++i;
				--j;
			} else if (vowels.contains(arr[i])) {
				--j;
			} else {
				++i;
			}
		}

		return new String(arr);
    }

    private void swap(char[] arr, int i, int j) {
        char temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }
}
